### ISSUE TITLE
[Bug Report] PCC drop observed in ttnn.softmax with large-value input tensors

### ISSUE THREAD

**Issue by kamalrajkannan78** [tenstorrent/tt-metal#25193]
### Description

When using ttnn.softmax on input tensors containing large values (torch.randn(1, 14, 39, 39) * 50 + 10), the output exhibits a significant PCC drop when compared to PyTorch's F.softmax

[jul16_softmax_failure_metal.log](https://github.com/user-attachments/files/21263556/jul16_softmax_failure_metal.log)

### Steps to reproduce the issue

```
git checkout kkannan/jul16_softmax_drop 
git submodule update --init --recursive
pytest tests/ttnn/unit_tests/operations/fused/test_softmax.py -k "my" -svv
```

### Observed behaviour 

```
for dim=-1
torch.randn(1, 14, 39, 39),  # PCC=0.9996815652237587
torch.randn(1, 14, 39, 39) * 50 + 10,  # PCC=0.291065779369528
```

### Expected behavior

ttnn.softmax should match PyTorch's F.softmax with high precision, even for input tensors with large or highly variable values

**Comment by kamalrajkannan78** [comment]
The previously problematic test case now having pcc = 0.9999487900570644 . 
Therefore, I’m closing this issue.

### Repro

```
git checkout kkannan/nov24_check_sanities
git submodule update --recursive
pytest tests/ttnn/unit_tests/operations/sanities/test_softmax.py -svv
```

### Logs

- [nov24_softmax.log](https://github.com/user-attachments/files/23701923/nov24_softmax.log)
